internal/resources/diffValidation/resourse_router.go

```diff
@@ -51,12 +52,18 @@ func (r *Router) validateTier0Config() error {
 	}
 
 	//BGP graceful restart timers cannot be updated when BGP config is enabled
-	bgpEnabledPath := "tier0_config.0.bgp.0.enable_bgp"
-	if r.diff.HasChange(bgpEnabledPath) || r.diff.HasChange("tier0_config.0.bgp.0.restart_time") || r.diff.HasChange("tier0_config.0.bgp.0.stale_route_time") {
-		action := r.diff.Get(bgpEnabledPath)
+	if r.diff.HasChange(utils.BgpEnabledPath) || (r.diff.HasChange(utils.BgpRestartTimerPath) || r.diff.HasChange(utils.BgpStaleTimerPath)) {
```

Comment:
sorry to go back and forth. I would recommend having constants specific to package in the package itself and anything global in a package with name 'Constants'. Its better to have utils having utility static methods.

Also please try to split this into multiple smaller pieces if possible. That would add more readability to the code.
For example:
validateTier0Config(){
    validateBGPrestart 
    validateBGPsr

these are just high level example name. Please use what ever makes more sense


}